url8e6440c8aa_robdimsdale_project_euler_tgz-pJ8-com_rmd_personal_projecteuler_MainJ8/src/com/rmd/personal/projecteuler/p20to39/P028.java

```diff
@@ -0,0 +1,109 @@
+package com.rmd.personal.projecteuler.p20to39; // SUPPRESS CHECKSTYLE fileLength
+
+import com.rmd.personal.projecteuler.Problem;
+
+public class P028 implements Problem {
+
+    private static final int DEFAULT_MAX_A = 1001;
+
+    private int squareSize;
+
+    private long[][] numberSquare;
+
+    public P028() {
+        this.setSquareSize(DEFAULT_MAX_A);
+    }
+
+    public int getSquareSize() {
+        return squareSize;
+    }
+
+    public void setSquareSize(int squareSize) {
+        this.squareSize = squareSize;
+    }
+
+    private long[][] getNumberSquare() {
+        return numberSquare;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Starting with the number 1 and moving to the right in a clockwise direction "
+                + "a 5 by 5 spiral is formed as follows:\n\n"
+                + "21 22 23 24 25\n"
+                + "20  7  8  9 10\n"
+                + "19  6  1  2 11\n"
+                + "18  5  4  3 12\n"
+                + "17 16 15 14 13\n\n"
+                + "It can be verified that the sum of the numbers on the diagonals is 101.\n\n"
+                + "What is the sum of the numbers on the diagonals in a 1001 by 1001 spiral formed in the same way?";
+    }
+
+    @Override
+    public String run() {
+        this.initializeNumberSquare();
+
+        long sum = 0;
+        for (int i = 0; i < this.getSquareSize(); i++) {
+            sum += this.getNumberSquare()[i][i];
+            sum += this.getNumberSquare()[this.getSquareSize() - 1 - i][i];
+        }
+        sum -= 1; // Remove the double-counted '1' in the middle.
+
+        return String.valueOf(sum);
+    }
+
+    private void initializeNumberSquare() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.